script/makeinstaller.sh

```diff
@@ -1,62 +1,81 @@
-command -v convert > /dev/null 2>&1 || { echo >&2 "imagemagick is required to build the installer. Aborting."; exit 1; }
-command -v inkscape > /dev/null 2>&1 || { echo >&2 "inkscape is required to build the installer. Aborting."; exit 1; }
 
 # For CI nightly builds generate a version number with commit hash
 STREAMLINK_VERSION=$(python setup.py --version)
 STREAMLINK_VERSION_PLAIN="${STREAMLINK_VERSION%%+*}"
 STREAMLINK_INSTALLER="${1:-"streamlink-${STREAMLINK_VERSION/\+/_}"}"
+STREAMLINK_PYTHON_VERSION=3.6.6
+STREAMLINK_ASSETS_REPO="streamlink/streamlink-assets"
+STREAMLINK_ASSETS_RELEASE="latest"
 
-# include the build number
 CI_BUILD_NUMBER=${GITHUB_RUN_ID:-0}
 STREAMLINK_VI_VERSION="${STREAMLINK_VERSION_PLAIN}.${CI_BUILD_NUMBER}"
 
-build_dir="$(pwd)/build"
+DIST_DIR="${STREAMLINK_DIST_DIR:-"${ROOT}/dist"}"
+INSTALLER_PATH="${DIST_DIR}/${STREAMLINK_INSTALLER}.exe"
+
+build_dir="${ROOT}/build"
 build_dir_plugins="${build_dir}/lib/streamlink/plugins"
+cache_dir="${build_dir}/cache"
 nsis_dir="${build_dir}/nsis"
 files_dir="${build_dir}/files"
 icons_dir="${files_dir}/icons"
-# get the dist directory from an environment variable, but default to the build/nsis directory
-dist_dir="${STREAMLINK_DIST_DIR:-${nsis_dir}}"
-mkdir -p "${build_dir}" "${dist_dir}" "${nsis_dir}" "${files_dir}" "${icons_dir}"
 
-echo "Building streamlink-${STREAMLINK_VERSION} package..." 1>&2
+removed_plugins_file="${ROOT}/src/streamlink/plugins/_removed.txt"
+
+log "Setting up clean build directories"
+[[ -d "${build_dir}" ]] && rm -rf "${nsis_dir}" "${files_dir}" "${icons_dir}"
+mkdir -p "${build_dir}" "${cache_dir}" "${nsis_dir}" "${files_dir}" "${icons_dir}" "${DIST_DIR}"
+
+
+log "Building streamlink-${STREAMLINK_VERSION} package"
 python setup.py build 1>&2
 
+
+log "Creating empty plugin files"
 # https://github.com/streamlink/streamlink/issues/1223
-echo "Create empty files."
-old_files=(
-  "afreecatv" "alieztv" "apac" "azubutv" "bambuser" "beam" "bliptv" "canlitv"
-  "connectcast" "cyro" "daisuki" "disney_de" "dmcloud" "dmcloud_embed"
-  "douyutv_blackbox" "filmon_us" "furstream" "gaminglive" "gomexp" "letontv"
-  "livecodingtv" "livestation" "looch" "media_ccc_de" "meerkat" "neulion"
-  "nineanime" "pcyourfreetv" "seemeplay" "servustv" "stream" "streamlive"
-  "streamupcom" "tv8cat" "ufctv" "veetle" "viagame" "viasat_embed" "wattv"
-  "aftonbladet" "aliez" "antenna" "arconai" "bongacams" "brittv" "cam4"
-  "camsoda" "chaturbate" "expressen" "mips" "seetv" "speedrunslive" "streamboat"
-  "vgtv" "weeb" "oldlivestream" "ok_live" "rte" "tvcatchup" "npo" "ovvatv"
-  "mlgtv" "kanal7" "younow" "eurocom" "tv1channel"
-)
-for i in "${old_files[@]}"
-do
-    touch "${build_dir_plugins}/$i.py"
-done
+while read -r pluginname; do
+    touch "${build_dir_plugins}/${pluginname}.py"
+done < <(sed -e 's/[[:space:]]*[#;].*//; /^[[:space:]]*$/d' "${removed_plugins_file}")
```

Comment:
I think it's a bit more readable to use `readarray`, it's a Bash 4 command but that shouldn't be a problem. 

```suggestion
readarray -t removed_plugins < "${removed_plugins_file}"
for pluginname in "${removed_plugins[@]}"; do
    touch "${build_dir_plugins}/${pluginname}.py"
done
```

Reply:
Does readarray remove comments with a leading `#` or `;`?

Reply:
True, you got comments in there uglying it up :p 
I made a similar change on another branch, but didn't put comments in the plugin list - I wanted to make a script to check to make sure a removed plugin was added to that file.

Reply:
I don't have anything against removing the comments. I just included them for anyone randomly reading the file.

> I wanted to make a script to check to make sure a removed plugin was added to that file

That script/test could also remove the comments while readin from the file. :smiley: 

But ok, let me change this, too.

Reply:
Nah, leave the comments they are useful. `grep -x` fixes everything.
```
while read -r rplugin; do echo grep -Fxq "${rplugin}" src/streamlink/plugins/_removed.txt; done < <(git diff --cached --name-only --diff-filter=D master | grep -oP '^src/streamlink/plugins/(\K.*)(?=.py)');
```

Reply:
Ok, restored again. For some reason Github doesn't update the review comments and I have to refresh the page.